### ISSUE TITLE
Breaks YouTube?

### ISSUE THREAD

**Issue by B00ze64** [snyderp/web-api-manager#83]
Good day.

Cannot find a way to get YouTube to work; as soon as I block ONE API, it does not matter which one, YouTube breaks. Guess the "Permitting Rules" are not totally useless after all...

Regards,

**Comment by CHEF-KOCH** [comment]
The bugreport is pretty pointless cause you not mentioned which API's you tried to block, I block 34 API's and it still works. 

```
HTML: Web Workers
DOM Parsing and Serialization
HTML: Canvas Element
Scalable Vector Graphics (SVG) 1.1 (Second Edition)
WebGL 2.0
WebGL Specification
Ambient Light Sensor API
Battery Status API
DeviceOrientation Events
Gamepad
Proximity Events
Screen Orientation API
Vibration API
WebUSB API
MediaStream Recording
Web Audio API
Web Speech API
WebVR
WebVTT: The Web Video Text Tracks Format
Geolocation API
HTML: Plugins
Beacon
HTML: Channel Messaging
HTML: Web Sockets
Push API
Tracking Preference Expression (DNT)
WebRTC 1.0: Real-time Communication Between Browser
Fullscreen API
UI Events Specification
Indexed Database API
High Resolution Time (Level 2)
Navigation Timing
Performance Timeline (Level 2)
Resource Timing
```

I think that's a Browser/implementation thingy here.

**Comment by B00ze64** [comment]
Yeah, I did forget to mention I use Firefox 52 ESR. As for which I block, as I stated, it does not matter; as soon as I block /something/ then the site breaks.

**Comment by CHEF-KOCH** [comment]
Can you post your Youtube settings?

**Comment by B00ze64** [comment]
Here's an example: I clear all settings in WebAPI Manager and open the YouTube homepage; everything's fine. Then I open the WebAPI Manager settings and block Light-Sensor. Switch tabs and reload YouTube and it's mostly blank - I have the search bar at the top, but nothing else. To be absolutely certain it's not some kind of interaction between other addOns and WebAPI Manager, I would have to load a fresh profile with only WebAPI Manager installed. I used to have such a profile but I deleted it about 2 weeks ago. I used it to test the saving of my settings in WebAPI Manager, and sure enough, WebAPI Manager forgot all its settings each time I restarted the browser (this is probably an issue with old FF52 and it's early support of WebExtensions (i.e. it probably does save settings correctly in newer versions of Firefox.)) So at least in FF52 and with a few more addOns active, YouTube breaks as soon as WebAPI Manager is active.

**Comment by CHEF-KOCH** [comment]
You need at least [FF 56](https://ftp.mozilla.org/pub/firefox/releases/56.0/) due WebExtension API change. 

I quickly checked it and 56 seems the oldest version which worked for me.

**Comment by B00ze64** [comment]
Hi. Well, the author marked WebAPI Manager as compatible with FF52, so I am hoping he wants it to be so, and can tweak around FF52 WebExtention bugs. The problem with newer FF's is they dropped support for XUL, and finding a replacement for my 65 active XUL addOns is going to take so much time, I am not going to do it anytime soon. But thanks for confirming it is a FF52 issue. Regards,
